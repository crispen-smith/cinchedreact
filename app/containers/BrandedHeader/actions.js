/*
 *
 * Brand actions
 *
 */

import BRAND_ACTIONS from './constants';

export function login() {
  return {
    type: BRAND_ACTIONS.LOGIN,
  };
}

export function logout() {
  return {
    type: BRAND_ACTIONS.LOGOUT,
  };
}
