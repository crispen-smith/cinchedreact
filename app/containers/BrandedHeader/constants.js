/*
 *
 * Header constants
 *
 */

import keymirror from 'keymirror';

const BRAND_ACTIONS = keymirror({
  LOGIN: null,
  LOGOUT: null,
});

export default BRAND_ACTIONS;
