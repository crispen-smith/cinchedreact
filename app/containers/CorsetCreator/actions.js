/*
 *
 * CorsetCreator actions
 *
 */

import actionTypes from './constants';

export function defaultAction() {
  return {
    type: actionTypes.default,
  };
}

export function create(corset) {
  return {
    type: actionTypes.create,
    corset,
  };
}

export function save() {
  return {
    type: actionTypes.save,
  };
}

export function setName(name) {
  return {
    type: actionTypes.setName,
    name,
  };
}

export function setType(corsetType) {
  return {
    type: actionTypes.setType,
    corsetType,
  };
}

export function setDescription(description) {
  return {
    type: actionTypes.setDescription,
    description,
  };
}

export function setPrimaryImage(primaryImageURL) {
  return {
    type: actionTypes.setPrimaryImage,
    primaryImageURL,
  };
}

export function addImage(imageURL) {
  return {
    type: actionTypes.addImage,
    imageURL,
  };
}

export function setPrice(price) {
  return {
    type: actionTypes.setPrice,
    price,
  };
}

/*
  All Corsets must be loaded to corsetGallery 
*/
