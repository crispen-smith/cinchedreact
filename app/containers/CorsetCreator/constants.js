/*
 *
 * CorsetCreator constants
 *
 */

import keymirror from 'keymirror';

const actionTypes = keymirror({
  default: null,
  save: null,
  isSaved: null,
  create: null,
  setDescription: null,
  setPrimaryImage: null,
  addImage: null,
  setPrice: null,
  reset: null,
});

export default actionTypes;
