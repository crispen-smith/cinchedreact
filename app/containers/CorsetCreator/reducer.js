/*
 *
 * CorsetCreator reducer
 *
 */

import { fromJS } from 'immutable';
import actionTypes from './constants';

export const initialState = fromJS({});

function corsetCreatorReducer(state = initialState, action) {
  switch (action.type) {
    case actionTypes.create:
      return createCorset(state, action);

    case actionTypes.save:
      return fromJS({
        ...state.toJS(),
        saving: true,
      });

    case actionTypes.isSaved:
      return setSaved(state);
    case actionTypes.setDescription:
      return setDescription(state, action);
    case actionTypes.setPrimaryImage:
      return setPrimaryImage(state, action);
    case actionTypes.addImage:
      return addImage(state, action);
    case actionTypes.setPrice:
      return setPrice(state, action);
    case actionTypes.reset:
      return fromJS({
        ...state.toJS(),
        currentCorset: null,
      });
    case actionTypes.default:
      return state;
    default:
      return state;
  }
}

const createCorset = (state, action) => {
  const newState = state.toJS();
  if (!newState.corsets) newState.corsets = [];
  newState.corsets.push(action.corset);
  newState.currentCorset = action.corset.name;
  return fromJS(newState);
};

/* eslint-disable no-underscore-dangle */
const __getCurrentCorset = state => {
  const { corsets } = state.toJS();
  const currentCorset = corsets.filter(
    corset => corset.name === state.toJS().currentCorset,
  );
  return currentCorset[0];
};

const __getNewCorsetList = (state, currentCorset) => {
  const corsetList = state.corsets.filter(
    corset => corset.name !== state.currentCorset,
  );

  corsetList.push(currentCorset);
  return corsetList;
};

/* eslint-enable */

const setDescription = (state, action) => {
  const newState = {
    ...state.toJS(),
  };
  const currentCorset = __getCurrentCorset(state);
  currentCorset.description = action.description;
  newState.corsets = __getNewCorsetList(newState, currentCorset);
  return fromJS(newState);
};

const setPrimaryImage = (state, action) => {
  const newState = {
    ...state.toJS(),
  };
  const currentCorset = __getCurrentCorset(state);
  currentCorset.primaryImage = action.image;
  newState.corsets = __getNewCorsetList(newState, currentCorset);
  return fromJS(newState);
};

const addImage = (state, action) => {
  const newState = {
    ...state.toJS(),
  };
  const currentCorset = __getCurrentCorset(state);
  if (!currentCorset.images) currentCorset.images = [];
  currentCorset.images.push(action.image);
  newState.corsets = __getNewCorsetList(newState, currentCorset);
  return fromJS(newState);
};

const setPrice = (state, action) => {
  const newState = {
    ...state.toJS(),
  };
  const currentCorset = __getCurrentCorset(state);
  currentCorset.price = action.price;
  newState.corsets = __getNewCorsetList(newState, currentCorset);
  return fromJS(newState);
};

const setSaved = state => {
  const newState = {
    ...state.toJS(),
    saving: false,
  };
  const currentCorset = __getCurrentCorset(state);
  currentCorset.saved = true;
  newState.corsets = __getNewCorsetList(newState, currentCorset);
  return fromJS(newState);
};

export default corsetCreatorReducer;

// const actionTypes = keymirror({
//  X default
//  X save
//  X isSaved
//  X create (Adds a new entry to the corsets array - expects a name and a type)
//  X setDescription
//  X setPrimaryImage
//  X addImage
//  X setPrice
// });
