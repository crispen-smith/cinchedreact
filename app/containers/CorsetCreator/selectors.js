import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the corsetCreator state domain
 */

const selectCorsetCreatorDomain = state =>
  state.get('corsetCreator', initialState);

/**
 * Other specific selectors
 */

/**
 * Default selector used by CorsetCreator
 */

const makeSelectCorsetCreator = () =>
  createSelector(selectCorsetCreatorDomain, substate => substate.toJS());

export default makeSelectCorsetCreator;
export { selectCorsetCreatorDomain };
