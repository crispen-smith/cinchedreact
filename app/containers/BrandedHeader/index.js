/**
 *
 * BrandedHeader
 *
 */

import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';

import { makeSelectLoggedIn } from './selectors';
import reducer from './reducer';
import saga from './saga';
import BrandBox from '../../components/BrandBox';
import LogToggle from '../../components/LogToggle';

import * as brandActions from './actions';
import { colors } from '../../resources/colors';

const BH = styled.div`
  display: flex;
  background-color: ${colors.pink};
  border-bottom: 1px solid rgba(255, 255, 255, 0.1);
  box-shadow: 1px 1px 1px 0px rgba(0, 0, 0, 0.1);
`;

class BrandedHeaderBase extends React.Component { // eslint-disable-line
  render() {
    const dispatch = { ...this.props };
    let action;
    if (this.props.loggedIn) {
      action = brandActions.logout();
    } else {
      action = brandActions.login();
    }

    const loggingSwitch = () => dispatch(action);
    return (
      <BH>
        <BrandBox />
        <LogToggle loggedIn={this.props.loggedIn} onClick={loggingSwitch} />
      </BH>
    );
  }
}

BrandedHeaderBase.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
};

const mapStateToProps = createStructuredSelector({
  loggedIn: makeSelectLoggedIn(),
});

export function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'brand', reducer });
const withSaga = injectSaga({ key: 'brand', saga });

const BrandedHeader = compose(
  withReducer,
  withSaga,
  withConnect,
)(BrandedHeaderBase);

export default BrandedHeader;
