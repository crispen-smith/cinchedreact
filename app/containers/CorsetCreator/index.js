/**
 *
 * CorsetCreator
 *
 */

import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Helmet } from 'react-helmet';
import { createStructuredSelector } from 'reselect';
import { Redirect } from 'react-router-dom';

import { compose } from 'redux';

import injectReducer from 'utils/injectReducer';
import {
  makeSelectCorsetGallery,
  makeSelectCurrentCorset,
  makeSelectorFilteredCorsets,
} from '../CorsetGallery/selectors';
import { makeSelectLoggedIn } from '../BrandedHeader/selectors';
import reducer from './reducer';
import * as actions from './actions';
import FormComponent from '../../components/FormComponent';
import NameBox from '../../components/NameBox';
import DropDown from '../../components/DropDown';
import SubmitButton from '../../components/SubmitButton';
import NotLoggedInComponent from '../../components/NotLoggedInComponent';

/* eslint-disable react/prefer-stateless-function */
export class CorsetCreator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      corsetGallery: props.corsetGallery,
      productName: '',
      productType: 'Overbust',
      enabled: false,
      created: false,
    };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleNameChange = this.handleNameChange.bind(this);
    this.handleProductChange = this.handleProductChange.bind(this);
    this.handleChange = changeHandler;
  }

  handleNameChange(e) {
    if (e.persist) e.persist();

    e.preventDefault();
    this.nameValueCallback = e.nameCallback
      ? e.nameCallback.bind(this)
      : () => {};
    this.nameEnabledCallBack = e.enabledCallBack
      ? e.enabledCallBack.bind(this)
      : () => {};
    this.nameChangeCallback = () => {
      this.nameValueCallback();
      this.setState(
        { enabled: this.handleChange(e.target.value, this.state) },
        this.nameEnabledCallBack,
      );
    };
    this.setState({ productName: e.target.value }, this.nameChangeCallback);
  }

  handleProductChange(e) {
    if (e.persist) e.persist();

    e.preventDefault();
    this.typeValueCallback = e.valueCallback
      ? e.valueCallback.bind(this)
      : () => {};
    this.typeEnabledCallback = e.enabledCallback
      ? e.enabledCallBack.bind(this)
      : () => {};
    this.setState({ productType: e.target.value });
    this.setState({ enabled: this.handleChange(e.target.value, this.state) });
  }

  handleSubmit(e) {
    if (e.persist) e.persist();

    e.preventDefault();
    this.action = actions.create({
      name: this.state.productName,
      type: this.state.productType,
    });
    this.props.dispatch(this.action);
    this.submitCallback = e.handler ? e.handler.bind(this) : () => {};
    this.setState({ created: true }, this.submitCallback);
  }

  render() {
    if (!this.props.loggedIn)
      return <NotLoggedInComponent title="Corset Creator" />;
    if (!this.state.created) {
      return (
        <Fragment>
          <Helmet>
            <title>Create a new Corset</title>
            <meta
              name="description"
              content="Use this screen to create a new corset in the corset gallery."
            />
          </Helmet>
          <h1>Create a new Corset</h1>
          <FormComponent>
            <NameBox
              inputName="productName"
              inputLabel="Name"
              onChange={this.handleNameChange}
              value={this.state.productName}
            />
            <DropDown
              inputName="productType"
              inputLabel="Type"
              options={['Overbust', 'Underbust']}
              value={this.state.productType}
              onChange={this.handleProductChange}
            />
            {this.state.enabled ? (
              <SubmitButton
                label="Create"
                enabled
                dataCursor="pointer"
                onClick={this.handleSubmit}
              />
            ) : (
              <SubmitButton
                label="---"
                enabled={false}
                dataCursor="not-allowed"
                onClick={this.handleSubmit}
              />
            )}
          </FormComponent>
        </Fragment>
      );
    }
    return (
      <Redirect
        to={`/corsets/edit/${this.state.productType}/${this.state.productName}`}
      />
    );
  }
}

export function changeHandler(e, state) {
  if (e === '' || state.productName === '') return false;

  const corsets =
    state.corsetGallery && state.corsetGallery.corsets
      ? state.corsetGallery.corsets
      : null;

  if (!corsets || corsets.length === 0 || !corsets.filter) return true;

  const productType =
    e === 'Underbust' || e === 'Overbust' ? e : state.productType;
  const productName =
    e === 'Underbust' || e === 'Overbust' ? state.productName : e;

  const filteredCorsets = corsets.filter(
    corset => corset.type === productType && corset.name === productName,
  );

  return !(filteredCorsets && filteredCorsets.length > 0);
}

CorsetCreator.propTypes = {
  dispatch: PropTypes.func.isRequired,
  loggedIn: PropTypes.bool.isRequired,
  corsetGallery: PropTypes.object.isRequired,
};

const mapStateToProps = createStructuredSelector({
  corsetGallery: makeSelectCorsetGallery(),
  currentCorset: makeSelectCurrentCorset(),
  filteredCorsets: makeSelectorFilteredCorsets(),
  loggedIn: makeSelectLoggedIn(),
});

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'corsetGallery', reducer });

export default compose(
  withReducer,
  withConnect,
)(CorsetCreator);
