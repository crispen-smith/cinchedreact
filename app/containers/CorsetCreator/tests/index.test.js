import React from 'react';

import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import configureStore from '../../../configureStore';
import CorsetCreator, { changeHandler } from '../index';
import BrandedHeader from '../../BrandedHeader';

describe('<CorsetCreator />', () => {
  it('Renders a CorsetCreator when loggedIn and currentCorset is empty', () => {
    const store = configureStore({}, browserHistory);

    const creator = mount(
      <Provider store={store}>
        <CorsetCreator />
      </Provider>,
    );

    expect(creator).toMatchSnapshot();
    expect(creator.find('DropDown')).toHaveLength(1);
    expect(creator.find('NameBox')).toHaveLength(1);
    expect(creator.find('SubmitButton')).toHaveLength(1);
  });

  it('Does not render the CorsetCreator when not logged in', () => {
    const store = configureStore({}, browserHistory);
    const creator = mount(
      <Provider store={store}>
        <div>
          <BrandedHeader />
          <CorsetCreator />
        </div>
      </Provider>,
    );
    const bhLink = creator.find('a');
    bhLink.simulate('click');

    expect(creator).toMatchSnapshot();
    expect(creator.find('DropDown')).toHaveLength(0);
    expect(creator.find('NameBox')).toHaveLength(0);
    expect(creator.find('SubmitButton')).toHaveLength(0);
  });

  it('Creates a new corset entry when the  submitButton is enabled and clicked', () => {
    const brand = { loggedIn: true };
    const corsetGallery = { created: true };
    const store = configureStore({}, browserHistory, brand, corsetGallery);

    const creator = mount(
      <Provider store={store}>
        <CorsetCreator />
      </Provider>,
    );

    const cc = creator.find('CorsetCreator').at(0);
    const nameBox = creator.find('NameBox').at(0);

    nameBox.simulate('change', {
      // First Call - with Handler defined
      target: {
        value: 'test',
        preventDefault: () => {},
      },
    });
    const submitButton = creator
      .find('SubmitButton')
      .at(0)
      .find('input')
      .at(0);

    submitButton.simulate('click', {
      // First Call - with Handler defined
      target: {
        preventDefault: () => {},
        handler: () => {
          const redirect = cc.find('redirect').at(0);
          expect(redirect).toBeDefined();
          expect(cc.instance().action).toBeDefined();
          expect(cc.instance().submitCallback).toBeDefined();
        },
      },
    });
  });

  it('handles name changes correctly', () => {
    const store = configureStore({}, browserHistory);
    const creator = mount(
      <Provider store={store}>
        <CorsetCreator />
      </Provider>,
    );

    const NameBox = creator
      .find('NameBox')
      .at(0)
      .find('input')
      .at(0);

    function nameCallback() {
      expect(this.state.productName).toEqual('test');
    }

    function enabledCallBack() {
      expect(this.state.enabled).toEqual(false);
    }

    NameBox.simulate('change', {
      target: { value: 'test' },
      preventDefault: () => {},
      nameCallback,
      enabledCallBack,
    });

    const cc = creator.find('CorsetCreator').at(0);

    expect(cc.instance().nameValueCallback).toBeDefined();
    expect(cc.instance().nameEnabledCallBack).toBeDefined();

    NameBox.simulate('change', {
      target: { value: 'test' },
      preventDefault: () => {},
    });

    expect(cc.instance().nameValueCallback).toBeDefined();
    expect(cc.instance().nameEnabledCallBack).toBeDefined();
  });

  it('handles ProductType changes correctly', () => {
    const store = configureStore({}, browserHistory);
    const creator = mount(
      <Provider store={store}>
        <CorsetCreator />
      </Provider>,
    );

    const productType = creator
      .find('DropDown')
      .at(0)
      .find('option')
      .at(0);

    function valueCallback() {
      expect(this.state.ProductType).toEqual('Underbust');
    }

    function enabledCallBack() {
      expect(this.state.enabled).toEqual(false);
    }

    productType.simulate('change', {
      target: { value: 'test' },
      preventDefault: () => {},
      valueCallback,
      enabledCallBack,
    });

    const cc = creator.find('CorsetCreator').at(0);

    expect(cc.instance().typeValueCallback).toBeDefined();
    expect(cc.instance().typeEnabledCallback).toBeDefined();

    productType.simulate('change', {
      target: { value: 'test' },
      preventDefault: () => {},
    });

    expect(cc.instance().typeValueCallback).toBeDefined();
    expect(cc.instance().typeEnabledCallback).toBeDefined();
  });
});

describe('ChangeHandler interactions with Name', () => {
  it('returns false when passed an empty value', () => {
    expect(
      changeHandler('', {
        corsetGallery: { corsets: [{ name: 'test', type: 'Underbust' }] },
      }),
    ).toEqual(false);
  });
  it('returns true when passed an empty set of corsets', () => {
    expect(
      changeHandler('test', {
        corsetGallery: { corsets: [] },
      }),
    ).toEqual(true);
  });
  it('returns true when passed a name/type tuple not in the array', () => {
    expect(
      changeHandler('test2', {
        corsetGallery: {
          corsets: [{ name: 'test', type: 'Underbust' }],
          productType: 'Underbust',
        },
      }),
    ).toEqual(true);
  });
});
describe('ChangeHandler interactions with ProductType', () => {
  it('returns false when passed any ProductType and any name in state', () => {
    expect(
      changeHandler('Underbust', {
        productName: '',
        corsetGallery: { corsets: [{ name: 'test', type: 'Underbust' }] },
      }),
    ).toEqual(false);
  });
  it('returns true when passed an empty set of corsets as a result of ', () => {
    expect(
      changeHandler('Underbust', {
        corsetGallery: {
          corsets: [],
          ProductName: 'Test',
        },
      }),
    ).toEqual(true);
  });
  it('returns true when passed a name/type tuple not in the array', () => {
    expect(
      changeHandler('Overbust', {
        corsetGallery: {
          corsets: [{ name: 'test', type: 'Underbust' }],
          productType: 'Underbust',
        },
      }),
    ).toEqual(true);
  });
});
