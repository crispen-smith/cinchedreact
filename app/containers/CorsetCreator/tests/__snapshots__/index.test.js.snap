// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CorsetCreator /> Does not render the CorsetCreator when not logged in 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {
        "brand": [Function],
        "corsetGallery": [Function],
      },
      "injectedSagas": Object {
        "brand": Object {
          "mode": "@@saga-injector/restart-on-remount",
          "saga": [Function],
          "task": Object {
            "@@redux-saga/TASK": true,
            "cancel": [Function],
            "cont": undefined,
            "done": Promise {},
            "error": [Function],
            "id": 1,
            "isAborted": [Function],
            "isCancelled": [Function],
            "isRunning": [Function],
            "joiners": null,
            "name": "defaultSaga",
            "result": [Function],
            "setContext": [Function],
          },
        },
      },
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <div>
    <withReducer(withSaga(Connect(BrandedHeaderBase)))>
      <withSaga(Connect(BrandedHeaderBase))>
        <Connect(BrandedHeaderBase)>
          <BrandedHeaderBase
            dispatch={[Function]}
            loggedIn={false}
          >
            <BrandedHeader__BH>
              <div
                className="BrandedHeader__BH-HxBPG kskhDL"
              >
                <BrandBox>
                  <BrandBox__BrandBoxBase>
                    <div
                      className="BrandBox__BrandBoxBase-bOpzwz dtwMcz"
                    >
                      <MegaLogo>
                        <MegaLogo__MegaLogoBase>
                          <div
                            className="MegaLogo__MegaLogoBase-gVmnHe fjUeHC"
                          >
                            <MegaLogo__C>
                              <span
                                className="MegaLogo__C-hBLyQD CnhnX"
                              >
                                C
                              </span>
                            </MegaLogo__C>
                            <MegaLogo__C>
                              <span
                                className="MegaLogo__C-hBLyQD iPmZWc"
                              >
                                T
                              </span>
                            </MegaLogo__C>
                          </div>
                        </MegaLogo__MegaLogoBase>
                      </MegaLogo>
                    </div>
                  </BrandBox__BrandBoxBase>
                </BrandBox>
                <BrandTitle>
                  <BrandTitle__BT>
                    <div
                      className="BrandTitle__BT-iyACyt doctaa"
                    >
                      <BrandTitle__BrandTitleBase>
                        <h1
                          className="BrandTitle__BrandTitleBase-fCqnon fsKsKs"
                        >
                          Cinched Tight
                        </h1>
                      </BrandTitle__BrandTitleBase>
                      <BrandTitle__BrandMessage>
                        <h2
                          className="BrandTitle__BrandMessage-bmovll bAcupH"
                        >
                          Celebrate Your Shape
                        </h2>
                      </BrandTitle__BrandMessage>
                    </div>
                  </BrandTitle__BT>
                </BrandTitle>
                <LogToggle
                  loggedIn={false}
                  onClick={[Function]}
                >
                  <LogToggle__LT
                    onClick={[Function]}
                  >
                    <a
                      className="LogToggle__LT-bRttfU iacwAB"
                      onClick={[Function]}
                    >
                      Log In
                    </a>
                  </LogToggle__LT>
                </LogToggle>
              </div>
            </BrandedHeader__BH>
          </BrandedHeaderBase>
        </Connect(BrandedHeaderBase)>
      </withSaga(Connect(BrandedHeaderBase))>
    </withReducer(withSaga(Connect(BrandedHeaderBase)))>
    <withReducer(Connect(CorsetCreator))>
      <Connect(CorsetCreator)>
        <CorsetCreator
          corsetGallery={Object {}}
          dispatch={[Function]}
          filteredCorsets={Array []}
          loggedIn={false}
        >
          <h1>
            Corset Creator
          </h1>
          <p>
            you must be logged in as an admin to access this page
          </p>
        </CorsetCreator>
      </Connect(CorsetCreator)>
    </withReducer(Connect(CorsetCreator))>
  </div>
</Provider>
`;

exports[`<CorsetCreator /> Renders a CorsetCreator when loggedIn and currentCorset is empty 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {
        "corsetGallery": [Function],
      },
      "injectedSagas": Object {},
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <withReducer(Connect(CorsetCreator))>
    <Connect(CorsetCreator)>
      <CorsetCreator
        corsetGallery={Object {}}
        dispatch={[Function]}
        filteredCorsets={Array []}
        loggedIn={true}
      >
        <div>
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
          >
            <SideEffect(NullComponent)
              defer={true}
              encodeSpecialCharacters={true}
              meta={
                Array [
                  Object {
                    "content": "Use this screen to create a new corset in the corset gallery.",
                    "name": "description",
                  },
                ]
              }
              title="Create a new Corset"
              titleAttributes={Object {}}
            >
              <NullComponent
                defer={true}
                encodeSpecialCharacters={true}
                meta={
                  Array [
                    Object {
                      "content": "Use this screen to create a new corset in the corset gallery.",
                      "name": "description",
                    },
                  ]
                }
                title="Create a new Corset"
                titleAttributes={Object {}}
              />
            </SideEffect(NullComponent)>
          </HelmetWrapper>
          <FormComponent>
            <FormComponent__FC>
              <form
                className="FormComponent__FC-jRyaDy hZINgg"
              >
                <NameBox
                  inputLabel="Name"
                  inputName="productName"
                  onChange={[Function]}
                  value=""
                >
                  <div
                    className="form-row-primary"
                  >
                    <label
                      htmlFor="productName"
                    >
                      Name
                      <input
                        id="productName"
                        onChange={[Function]}
                        type="text"
                      />
                    </label>
                  </div>
                </NameBox>
                <DropDown
                  inputLabel="Type"
                  inputName="productType"
                  onChange={[Function]}
                  options={
                    Array [
                      "Overbust",
                      "Underbust",
                    ]
                  }
                  value="Overbust"
                >
                  <div
                    className="form-row-primary"
                  >
                    <label
                      htmlFor="productType"
                    >
                      Type
                      <select
                        id="productType"
                        onChange={[Function]}
                      >
                        <option
                          key="Overbust"
                          value="Overbust"
                        >
                          Overbust
                        </option>
                        <option
                          key="Underbust"
                          value="Underbust"
                        >
                          Underbust
                        </option>
                      </select>
                    </label>
                  </div>
                </DropDown>
                <SubmitButton
                  dataCursor="not-allowed"
                  enabled={false}
                  label="---"
                  onClick={[Function]}
                >
                  <SubmitButton__SB
                    data-cursor="not-allowed"
                    disabled={true}
                    onClick={[Function]}
                    type="submit"
                    value="---"
                  >
                    <input
                      className="SubmitButton__SB-cMjXsV hdYEGO"
                      data-cursor="not-allowed"
                      disabled={true}
                      onClick={[Function]}
                      type="submit"
                      value="---"
                    />
                  </SubmitButton__SB>
                </SubmitButton>
              </form>
            </FormComponent__FC>
          </FormComponent>
        </div>
      </CorsetCreator>
    </Connect(CorsetCreator)>
  </withReducer(Connect(CorsetCreator))>
</Provider>
`;
